File: Web/src/lib/client/Timers/GetTimerState.ts
import { createEndpoint } from "../api";
export type GetTimerStateRequest = void;
export type GetTimerStateResponse = {
    readonly remainingSeconds: number;
    readonly totalSeconds: number;
    readonly isRunning: boolean;
};
export const GetTimerState = createEndpoint<GetTimerStateRequest, GetTimerStateResponse>('/api/timers', 'GET');

File: API/Features/Timers/Endpoints/GetTimerState.cs
using API.Domain;
using Microsoft.AspNetCore.Http.HttpResults;

namespace API.Features.Timers.Endpoints;

public class GetTimerState
{
    public readonly record struct Response(int RemainingSeconds, int TotalSeconds, bool IsRunning);
    public static Results<Ok<Response>, ProblemHttpResult> Handle(IGameTimer timer)
    {
        if (timer.CurrentState == TimerState.Inactive)
        {
            return APIResults.NotFound("No active timer exists.");
        }

        return APIResults.Ok(new Response(
            (int)timer.RemainingTime.TotalSeconds,
            (int)timer.TotalTime.TotalSeconds,
            timer.CurrentState == TimerState.Running));
    }
}
